core/src/HIP/Kokkos_HIP_DeepCopy.cpp

```diff
@@ -41,7 +42,8 @@ void DeepCopyHIP(void* dst, void const* src, size_t n) {
 void DeepCopyAsyncHIP(const HIP& instance, void* dst, void const* src,
                       size_t n) {
   KOKKOS_IMPL_HIP_SAFE_CALL(
-      hipMemcpyAsync(dst, src, n, hipMemcpyDefault, instance.hip_stream()));
+      (instance.impl_internal_space_instance()->hip_memcpy_async_wrapper(
```

Comment:
Why did we add extra parentheses?

Reply:
Actually, I'm not reproducing it... I'm thinking a) I only saw this on the Cuda side of things or b) it was a relic from when I use using a function pointer impl. Let me test removing all of them and I'll push the change.